netcore/tests/Koralium.SqlToExpression.Tests/QueryTests.cs

```diff
@@ -1022,6 +1022,69 @@ public async Task TestInPredicateString()
             AssertAreEqual(expected, result.Result);
         }
 
+        [Test]
+        public async Task TestInPredicateParameters()
+        {
+            var parameters = new SqlParameters().Add(SqlParameter.Create("P0", "5-LOW"));
+            var result = await SqlExecutor.Execute("SELECT Orderkey, Orderpriority FROM \"order\" WHERE Orderpriority in (@P0)", parameters);
+
+            var expectedList = new List<string> { "5-LOW" };
+            var expected = TpchData.Orders
+                .Where(x => expectedList.Contains(x.Orderpriority))
+                .Select(x => new { x.Orderkey, x.Orderpriority })
+                .AsQueryable();
+
+            AssertAreEqual(expected, result.Result);
+        }
+
+        [Test]
+        public void TestInPredicateParameterMissing()
+        {
+            Assert.That(async () =>
```

Comment:
I have been testing async exceptions like this in Webeco:
```csharp
 Assert.ThrowsAsync<Exception>(() => _service.GetProfitOverview(searchModel, It.IsAny<OutputType>()));
``` 
I think you should be able to do something like this:
```csharp
var ex = Assert.ThrowsAsync<SqlErrorException>(async () => await SqlExecutor.Execute("SELECT Orderkey, Orderpriority FROM \"order\" WHERE Orderpriority in (@P0)"));
Assert.Equals("Could not find a parameter named: '@P0'", ex.Message);
```
 I think that one is a bit easier to read. Feel free to test if you want to, both works :)

Reply:
Changed to ThrowsAsync